crawl4ai/link_preview.py

```
@@ -0,0 +1,395 @@
+    
+    async def _extract_with_progress(
+        self, 
+        urls: List[str], 
+        seeding_config: SeedingConfig, 
+        link_config: Dict[str, Any]
+    ) -> List[Dict[str, Any]]:
+        """Extract head content with progress reporting."""
+        
+        total_urls = len(urls)
+        concurrency = link_config.concurrency
+        batch_size = max(1, total_urls // 10)  # Report progress every 10%
```

Ruff (0.14.0)

Local variable `batch_size` is assigned to but never used

Remove assignment to unused variable `batch_size`

(F841)